airbyte-scheduler/scheduler-persistence/src/main/java/io/airbyte/scheduler/persistence/job_error_reporter/SentryExceptionHelper.java

```diff
@@ -166,4 +179,130 @@ private static Optional<List<SentryException>> buildJavaSentryExceptions(final S
     return Optional.of(sentryExceptions);
   }
 
+  private static Optional<List<SentryException>> buildNormalizationDbtSentryExceptions(final String stacktrace) {
+    final List<SentryException> sentryExceptions = new ArrayList<>();
+
+    Map<String, String> usefulErrorMap = getUsefulErrorMessageAndTypeFromDbtError(stacktrace);
+
+    // if our errorMessage from the function != stacktrace then we know we've pulled out something
+    // useful
+    if (!usefulErrorMap.get(ERROR_MAP_MESSAGE_KEY).equals(stacktrace)) {
+      final SentryException usefulException = new SentryException();
+      usefulException.setValue(usefulErrorMap.get(ERROR_MAP_MESSAGE_KEY));
+      usefulException.setType(usefulErrorMap.get(ERROR_MAP_TYPE_KEY));
+      sentryExceptions.add(usefulException);
+    }
+
+    if (sentryExceptions.isEmpty())
+      return Optional.empty();
+
+    return Optional.of(sentryExceptions);
+  }
+
+  public static Map<String, String> getUsefulErrorMessageAndTypeFromDbtError(String stacktrace) {
+    // the dbt 'stacktrace' is really just all the log messages at 'error' level, stuck together.
+    // therefore there is not a totally consistent structure to these,
+    // see the docs: https://docs.getdbt.com/guides/legacy/debugging-errors
+    // the logic below is built based on the ~450 unique dbt errors we encountered before this PR
+    // and is a best effort to isolate the useful part of the error logs for debugging and grouping
+    // and bring some semblance of exception 'types' to differentiate between errors.
+    Map<String, String> errorMessageAndType = new HashMap<>();
```

Comment:
nit: the keys `errorMessage` and `errorType` could be an enum to type these `errorMessageAndType`-like maps more strictly

Reply:
good idea 👍